src/transformers/tokenization_utils_base.py

```diff
@@ -1580,20 +1628,18 @@ def _from_pretrained(cls, pretrained_model_name_or_path, *init_inputs, **kwargs)
             )
 
         # Save inputs and kwargs for saving and re-loading with ``save_pretrained``
-        tokenizer.init_inputs = init_inputs
-        tokenizer.init_kwargs = init_kwargs
+        # Removed: Now done at the base class level
+        # tokenizer.init_inputs = init_inputs
+        # tokenizer.init_kwargs = init_kwargs
 
         # If there is a complementary special token map, load it
         special_tokens_map_file = resolved_vocab_files.pop("special_tokens_map_file", None)
         if special_tokens_map_file is not None:
             with open(special_tokens_map_file, encoding="utf-8") as special_tokens_map_handle:
                 special_tokens_map = json.load(special_tokens_map_handle)
 
+            special_tokens_map = convert_added_tokens(special_tokens_map)
             for key, value in special_tokens_map.items():
-                if isinstance(value, dict):
```

Comment:
@thomwolf - this change currently breaks the `RagTokenizer`. 
If one runs the test: 
```
tests/test_modeling_rag.py::RagModelIntegrationTests::test_rag_sequence_generate_batch
```

and puts a breakpoint before `convert_added_tokens`, one can see why:

Previously a dict object corresponding *e.g.* to the BOS token, such as 
```
'bos_token': {'content': '<s>', 'single_word': False, 'lstrip': False, 'rstrip': False, 'normalized': True}
````

would have been processed by the line

```
value = AddedToken(**value)
```

Now such a dict (because it has no `__type` attribute is processed differently, which leads to errors later.
Not 100% sure how to solve it here. Do you have any good ideas: @LysandreJik @thomwolf @n1t0 ?